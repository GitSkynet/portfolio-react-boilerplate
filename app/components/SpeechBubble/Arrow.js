import styled from 'styled-components';
import PropTypes from 'prop-types';

const Arrow = styled.svg.attrs({
  xmlns: 'http://www.w3.org/2000/svg',
  viewBox: '0 0 848 640',
  preserveAspectRatio: 'xMinYMid',
})`
  float: ${({ isLeftHanded }) => isLeftHanded ? 'left' : 'right'};
  height: 1px;
  margin: -1px 0 0 0 !important;
  transform: rotateY(${({ isLeftHanded }) => isLeftHanded ? '0deg' : '180deg'});
  z-index: 9;

  &::after {
    content: "" !important;
    display: block !important;
    clear: both !important;
  }

  @media (min-height: ${({ showBreakpoint, height }) => showBreakpoint || height}) {
    height: ${({ height }) => height || '100%'};
  }
`;

Arrow.propTypes = {
  height: PropTypes.string,
  isLeftHanded: PropTypes.bool,
  showBreakpoint: PropTypes.string,
  width: PropTypes.string,
};

export default Arrow;
