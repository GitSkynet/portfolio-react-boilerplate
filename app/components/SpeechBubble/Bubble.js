import styled from 'styled-components';
import PropTypes from 'prop-types';

const Bubble = styled.div`
  display: inline-block;
  background-color: ${({ backgroundColor }) => backgroundColor || 'inherit'};
  border-radius: 0.5rem;
  box-shadow: 0 0 10px grey;
  color: ${({ color }) => color || 'inherit'};
  padding: ${({ padding }) => padding || 0};

  @media (min-height: ${({ showArrowBreakpoint }) => showArrowBreakpoint || '100%'}) {
    margin-right: 0;
    border-bottom-left-radius: ${(props) => (props.isLeftHanded) ? 0 : '0.5rem'};
    border-bottom-right-radius: ${(props) => (props.isLeftHanded) ? '0.5rem' : 0};
    box-shadow: none;
  }
`;

Bubble.propTypes = {
  backgroundColor: PropTypes.string,
  color: PropTypes.string,
  isLeftHanded: PropTypes.bool,
  padding: PropTypes.string,
  showArrowBreakpoint: PropTypes.string,
};

export default Bubble;
