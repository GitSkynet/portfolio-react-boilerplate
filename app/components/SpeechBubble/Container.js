import styled from 'styled-components';
import PropTypes from 'prop-types';

const Container = styled.div`
  z-index: 10;
  filter: none;

  @media (min-height: ${({ showArrowBreakpoint }) => showArrowBreakpoint || '100%'}) {
    filter: drop-shadow(0 0 5px grey);
  }

  @media (min-width: ${({ maxWidth }) => maxWidth || 'unset'}) {
    width: ${({ maxWidth }) => maxWidth || 'unset'};
  }
`;

Container.propTypes = {
  maxWidth: PropTypes.string,
  showArrowBreakpoint: PropTypes.string,
};

export default Container;
