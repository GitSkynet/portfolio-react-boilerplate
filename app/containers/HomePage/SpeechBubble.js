import styled from 'styled-components';
import { rem } from 'polished';

const SpeechBubble = styled.div`
  align-items: center;
  display: flex;
  height: 100%;
  filter: drop-shadow(0 0 10px grey);
  justify-content: center;
  padding: 10px 10px 10px 0; /* Keep room for shadow */

  @media (min-width: 600px) {
    padding-right: ${rem('114px')}; /* Keep room for site navigation */
  }
`;

export default SpeechBubble;
