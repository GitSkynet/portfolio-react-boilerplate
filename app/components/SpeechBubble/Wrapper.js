import styled from 'styled-components';
import { rem } from 'polished';

import Bubble from './Bubble';

const Wrapper = styled.div`
  margin-bottom: 9vh;
  padding: 1vh 1vw;
  width: 100%;
  filter: drop-shadow(0 0 10px grey);

  @media (min-width: 1008px) {
    max-width: ${rem('1008px')};
  }

  ${Bubble}

  p {
    &:not(:first-child) {
      margin-top: 1.5rem;
    }
  }
`;

export default Wrapper;
