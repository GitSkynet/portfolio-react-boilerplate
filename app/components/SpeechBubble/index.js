import React from 'react';
import PropTypes from 'prop-types';
import { lighten } from 'polished';

import Arrow from './Arrow';
import Bubble from './Bubble';
import Container from './Container';

class SpeechBubble extends React.PureComponent {
  render() {
    const {
      arrowHeight,
      backgroundColor,
      children,
      isLeftHanded,
      makeAppear,
      maxWidth,
      padding,
      showArrowBreakpoint,
    } = this.props;
    const lighterBackgroundColor = (backgroundColor && lighten(0.1, backgroundColor)) || 'inherit';

    return (
      <Container
        showArrowBreakpoint={showArrowBreakpoint}
        maxWidth={maxWidth}
        makeAppear={makeAppear}
      >
        <Bubble
          backgroundColor={backgroundColor}
          isLeftHanded={isLeftHanded}
          padding={padding}
          showArrowBreakpoint={showArrowBreakpoint}
        >
          {children}
        </Bubble>
        <Arrow
          isLeftHanded={isLeftHanded}
          height={arrowHeight}
          showBreakpoint={showArrowBreakpoint}
        >
          <path
            stroke={lighterBackgroundColor}
            fill={lighterBackgroundColor}
            d="M0 427.9l175.1 65.7-43.7-65.7z"
          />
          <path
            fill={backgroundColor}
            d="M175.1 493.6l-87.5 21.9v-54.8"
          />
          <path
            fill={lighterBackgroundColor}
            d="M87.6 515.5l21.2 44.5v-49.8"
          />
        </Arrow>
      </Container>
    );
  }
}

SpeechBubble.propTypes = {
  arrowHeight: PropTypes.string,
  backgroundColor: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  isLeftHanded: PropTypes.bool,
  makeAppear: PropTypes.bool,
  maxWidth: PropTypes.string,
  padding: PropTypes.string,
  showArrowBreakpoint: PropTypes.string,
};

export default SpeechBubble;
