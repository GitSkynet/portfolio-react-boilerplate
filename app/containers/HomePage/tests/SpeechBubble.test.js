import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import SpeechBubble from '../SpeechBubble';

describe('<SpeechBubble />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = shallow(<SpeechBubble />);
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
