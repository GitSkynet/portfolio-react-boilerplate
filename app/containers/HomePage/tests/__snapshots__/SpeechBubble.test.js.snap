// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpeechBubble /> should render and match the snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-filter: drop-shadow(0 0 10px grey);
  filter: drop-shadow(0 0 10px grey);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 10px 10px 10px 0;
}

@media (min-width:600px) {
  .c0 {
    padding-right: 7.125rem;
  }
}

<div
  className="c0"
/>
`;
